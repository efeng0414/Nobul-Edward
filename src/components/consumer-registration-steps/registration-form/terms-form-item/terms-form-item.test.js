import React from "react";
import TermsFormItem from "./index";
import { IntlProvider } from "react-intl";

describe("TermsFormItem Component", () => {
  const componentProps = {
    getFieldDecorator: fieldName => jsx => [fieldName, jsx],
    intl: global.getIntl(),
    termsFieldName: "terms",
    consentFieldName: "consent",
    consentForNewsValue: "consentForNews",
    termsValidator: () => {}
  };

  const componentJSX = (
    <IntlProvider {...componentProps.intl}>
      <TermsFormItem {...componentProps} />
    </IntlProvider>
  );

  describe("Component Snapshot", () => {
    const componentSnapshot = global.renderer.create(componentJSX).toJSON();

    it("should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });
});
