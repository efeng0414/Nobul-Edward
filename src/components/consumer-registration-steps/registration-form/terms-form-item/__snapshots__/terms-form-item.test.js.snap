// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsFormItem Component Component Snapshot should match stored snapshot 1`] = `
<div
  className="consumer-registration-form-item consumer-registration-form-checkboxes"
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          terms
          <div
            className="ant-checkbox-group"
          >
            <div
              className="consumer-registration-accept-terms-group"
            >
              <label
                className="ant-checkbox-wrapper"
              >
                <span
                  className="ant-checkbox"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-checkbox-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="checkbox"
                    value="accept_term"
                  />
                  <span
                    className="ant-checkbox-inner"
                  />
                </span>
                <span>
                  <div>
                    By checking this box you confirm that you have read and agree to the 
                    <span
                      className="accept-terms-link"
                    >
                      <a
                        href="https://corp.nobul.com/legal/terms-and-conditions/"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Terms of Service
                      </a>
                    </span>
                     and that you have read our Privacy Policy.
                  </div>
                </span>
              </label>
            </div>
          </div>
          consent
          <div
            className="ant-checkbox-group"
          >
            <div
              className="consumer-registration-accept-terms-group"
            >
              <label
                className="ant-checkbox-wrapper"
              >
                <span
                  className="ant-checkbox"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-checkbox-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="checkbox"
                    value="consentForNews"
                  />
                  <span
                    className="ant-checkbox-inner"
                  />
                </span>
                <span>
                  <div>
                    I consent to receiving news and updates about nobulâ€™s products and services. More information can be found in our 
                    <span
                      className="accept-terms-link"
                    >
                      <a
                        href="https://corp.nobul.com/legal/privacy-policy/"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Privacy Policy
                      </a>
                    </span>
                     or 
                    <span
                      className="accept-terms-link"
                    >
                      <a
                        href="https://corp.nobul.com/contact-us/"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Contact Us
                      </a>
                    </span>
                    . You may withdraw consent or unsubscribe at any time.
                  </div>
                </span>
              </label>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
